app/src/app/worker_schema/hybrid.rs

```diff
@@ -1,113 +1,64 @@
-
-use super::{
-    UseWorkerSchemaParserWithCache, WorkerSchemaApp, WorkerSchemaCacheHelper,
-    WorkerSchemaWithDescriptor,
-};
-
 // TODO use redis as cache ? (same implementation as rdb now)
 #[derive(Clone, DebugStub)]
 pub struct HybridWorkerSchemaAppImpl {
     storage_config: Arc<StorageConfig>,
-    id_generator: Arc<IdGeneratorWrapper>,
     #[debug_stub = "AsyncCache<Arc<String>, Vec<WorkerSchema>>"]
     async_cache: AsyncCache<Arc<String>, Vec<WorkerSchema>>,
-    memory_cache: MemoryCacheImpl<Arc<String>, WorkerSchemaWithDescriptor>,
+    descriptor_cache: Arc<MemoryCacheImpl<Arc<String>, WorkerSchemaWithDescriptor>>,
     repositories: Arc<HybridRepositoryModule>,
+    key_lock: Arc<RwLockWithKey<Arc<String>>>,
 }
 
 impl HybridWorkerSchemaAppImpl {
     pub fn new(
         storage_config: Arc<StorageConfig>,
-        id_generator: Arc<IdGeneratorWrapper>,
         memory_cache_config: &MemoryCacheConfig,
         repositories: Arc<HybridRepositoryModule>,
+        descriptor_cache: Arc<MemoryCacheImpl<Arc<String>, WorkerSchemaWithDescriptor>>,
     ) -> Self {
         Self {
             storage_config,
-            id_generator,
             async_cache: memory::new_memory_cache(memory_cache_config),
-            memory_cache: MemoryCacheImpl::new(memory_cache_config, None),
+            descriptor_cache,
             repositories,
+            key_lock: Arc::new(RwLockWithKey::new(memory_cache_config.max_cost as usize)),
         }
     }
 }
 
-impl UseIdGenerator for HybridWorkerSchemaAppImpl {
-    fn id_generator(&self) -> &IdGeneratorWrapper {
-        &self.id_generator
-    }
-}
 impl UseWorkerSchemaParserWithCache for HybridWorkerSchemaAppImpl {
-    fn cache(&self) -> &MemoryCacheImpl<Arc<String>, WorkerSchemaWithDescriptor> {
-        &self.memory_cache
+    fn descriptor_cache(&self) -> &MemoryCacheImpl<Arc<String>, WorkerSchemaWithDescriptor> {
+        &self.descriptor_cache
     }
 }
 
 #[async_trait]
 impl WorkerSchemaApp for HybridWorkerSchemaAppImpl {
-    async fn create_worker_schema(
-        &self,
-        worker_schema: WorkerSchemaData,
-    ) -> Result<WorkerSchemaId> {
-        let id = WorkerSchemaId {
-            value: self.id_generator().generate_id()?,
-        };
-        let db = self.worker_schema_repository().db_pool();
-        let mut tx = db.begin().await.map_err(JobWorkerError::DBError)?;
-        let id = self
-            .worker_schema_repository()
-            .create(&mut *tx, id, &worker_schema)
-            .await?;
-        tx.commit().await.map_err(JobWorkerError::DBError)?;
+    async fn load_worker_schema(&self) -> Result<bool> {
+        self.worker_schema_repository().add_from_plugins().await?;
         let _ = self
             .delete_cache_locked(&Self::find_all_list_cache_key())
             .await;
-
-        Ok(id)
+        Ok(true)
     }
-
-    // offset, limit 付きのリストキャッシュは更新されないので反映に時間かかる (ttl短かめにする)
-    async fn update_worker_schema(
```

Comment:
disable to create and update by request (fixed builtin and loaded plugins from file only)